Automattic/wp-calypso
# Jetpack plugin install: Move installer out of thank you modal

Implementing paObgF-fw-p2

## Changes

- Move the component responsible for provisioning plan plugins from the modal to the checklist. This ensures that the component remains mounted when the modal is dismissed or closed allowing for installation to proceed as expected.
- Modify installer to read desired plugin provisioning from query params. #34183 (merged) requests `all` installation after purchasing a plan which matches current behavior.

## Testing

* Checkout flows and My Plan pages should be unaffected for Simple and Atomic sites.
* Set up a Jetpack site on the free plan. The experience should be unchanged.
* Purchase a plan for the Jetpack site. The plan should be correctly provisioned as before (Akismet + VaultPress set up as required).
* Deactivate, remove, or delete plugin keys for Akismet and/or VaultPress on the Jetpack site. Visit `/plans/my-plan/SITE_SLUG?install=INSTALL_VALUE` with different `INSTALL_VALUE`s. Each should perform as expected:
  - `all`
  - `akismet`
  - `vaultpress`
  - `?install=akismet&install=vaultpress` should also work as expected.
  - `?install` or `?install=some-bad-value` should have no negative impact.
* Notice that the modal is no longer required for plugin provisioning to happen.

A [review](https://github.com/Automattic/wp-calypso/pull/34180#pullrequestreview-254896044) discovered that plugin keys may not initially be available, D31235-code is a patch to help testing that case.
1. Apply D31235-code and sandbox `public-api.wordpress.com`
1. Visit `/plans/my-plan/SITE_SLUG?thank-you&install=all`. You'll see repeated network requests for the keys until both are available.
1. Use the changes from D31235-code to make more keys available one at a time or all at once.
1. Plugins should begin provisioning as soon as they have a key if they've been requested.

In this case, progress still may not be reported accurately, but I don't believe that's a regression. I believe if both keys were not available before this patch, installation would never begin. Accurate progress reporting can be improved in a follow-up PR. I'd also suggest removing the progress in the modal and using a primary button to dismiss. It's no longer important for users to remain on the modal with the changes included here.
